pandas/core/tools/datetimes.py

```diff
@@ -200,19 +200,31 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
         if format is not None:
             raise ValueError("cannot specify both format and unit")
         arg = getattr(arg, 'values', arg)
+        tz_parsed = None
         result = tslib.array_with_unit_to_datetime(arg, unit,
                                                    errors=errors)
+        # GH 25546:
+        # Capture tz information from array_with_unit_to_datetime
+        if isinstance(result, tuple):
+            result, tz_parsed = result
         if box:
             if errors == 'ignore':
                 from pandas import Index
                 result = Index(result, name=name)
-                # GH 23758: We may still need to localize the result with tz
-                try:
-                    return result.tz_localize(tz)
-                except AttributeError:
+                if not isinstance(result, DatetimeIndex):
                     return result
-
-            return DatetimeIndex(result, tz=tz, name=name)
+            else:
```

Comment:
this is really really complicated, pls see if you can simplify. there are too many cases here.

Reply:
I similfied this as much as I could. The tricky thing here is that we need to evaluate the parsed tz first and then the passed `utc` argument next, taking into account whether the result is tz-aware after each step.